release-note-script/src/main/java/MergeReleaseNotes.java

```diff
@@ -0,0 +1,240 @@
+            category = Category.getByDisplayName(matcher.group(1));
+          continue;
+        }
+
+        matcher = Pattern.compile("^ *- *(\\p{Print}+)$").matcher(line);
+        if (matcher.matches()) {
+          if (category == null)
+            throw new IllegalStateException(
+                "Missing category. Release note text: " + matcher.group(1));
+          addReleaseNote(new ReleaseNote(edition, category, repository, matcher.group(1)));
+        }
+      }
+    }
+  }
+
+  private void addReleaseNote(ReleaseNote releaseNote) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap =
+        editionMap.computeIfAbsent(releaseNote.edition, k -> new EnumMap<>(Category.class));
+
+    Map<Repository, List<ReleaseNote>> repositoryMap =
+        categoryMap.computeIfAbsent(releaseNote.category, k -> new EnumMap<>(Repository.class));
+
+    List<ReleaseNote> releaseNotesList =
+        repositoryMap.computeIfAbsent(releaseNote.repository, k -> new ArrayList<>());
+
+    if (releaseNote.edition.equals(Edition.ENTERPRISE)) {
+      Matcher matcher =
+          Pattern.compile("(.*) +(\\((#[0-9]+ *)+\\))$").matcher(releaseNote.releaseNoteText);
+      if (matcher.matches()) {
+        if (DEBUG != null)
+          System.err.printf(
+              "Matched::%s::%s grp1:%s grp2:%s%n",
+              releaseNote.category, releaseNote.repository, matcher.group(1), matcher.group(2));
+        releaseNote.releaseNoteText = matcher.group(1);
+      }
+    }
+
+    releaseNotesList.add(releaseNote);
+    repositoryMap.put(releaseNote.repository, releaseNotesList);
+    categoryMap.put(releaseNote.category, repositoryMap);
+    editionMap.put(releaseNote.edition, categoryMap);
+  }
+
+  void output() {
+    System.out.print("## Summary\n\n");
+    Arrays.stream(Edition.values())
+        .forEach(
+            edition -> {
+              outputSections(edition);
+              System.out.println();
+            });
+  }
+
+  private void outputSections(Edition edition) {
+    Map<Category, Map<Repository, List<ReleaseNote>>> categoryMap = editionMap.get(edition);
+    if (categoryMap == null || categoryMap.isEmpty()) return;
+
+    System.out.printf("## %s edition%n", edition.getEdition());
+    Arrays.stream(Category.values())
+        .forEach(category -> outputReleaseNotes(category, categoryMap.get(category)));
+  }
+
+  private void outputReleaseNotes(
+      Category category, Map<Repository, List<ReleaseNote>> repositoryMap) {
+    if (repositoryMap == null || repositoryMap.isEmpty()) return;
+
+    System.out.printf("### %s%n", category.getDisplayName());
+    Arrays.stream(Repository.values())
+        .forEach(
+            repository -> {
+              List<ReleaseNote> releaseNotes = repositoryMap.get(repository);
+              if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                if (!repository.equals(Repository.DB))
+                  System.out.printf("#### %s%n", repository.getDisplayName());
```

Comment:
Why don't we need a h4 header for ScalarDB?

Reply:
As you can see this [example](https://github.com/scalar-labs/actions/blob/add-releasenote-script/release-note-script/src/test/resources/expected.md), the merged release note body consists of community edition part and enterprise edition part. The community edition part represents ScalarDB's release note body, meanwhile the enterprise edition represents the rest of repositories release note body. The enterprise edition part shows repository information (`ScalarDB Cluster`, `ScalarDB GraphQL`, `ScalarDB SQL`) under each category section. However, it doesn't need to show repository info for community edition because its only includes ScalarDB repository only.

 That is the reason ScalarDB doesn't need a h4 header.

Reply:
> Is it already described somewhere?

Currently no. I add some comments to describe it. Thank you.

Reply:
Thank you for the confirmation. Done at b14442b